### PR TITLE
fix: refacto components

### PR DESCRIPTION
**Pull request by guipasmoi**
image

### CODE REVIEW

File: app/routes/screens/main/clubFeed/ClubFeed.js [hunk 1]
```diff
@@ -44,23 +47,9 @@ class ClubFeed extends Component {
...
   render() {
     const { club, clubMembers, activities } = this.props;
-    if (club === Loading)
-      return (
-        <View style={styles.container}>
-          <Icon name="cached" color="#FC4C02" size={50} />
-          <Text style={styles.text}>fetching data</Text>
-        </View>
-      );
+    if (club === IsLoading) return <Loading />;
```
Comment (totorototo) [inline app/routes/screens/main/clubFeed/ClubFeed.js:50]: 👍

File: app/components/cardList/CardList.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Card, Icon } from "react-native-elements";
```
Comment (totorototo) [inline app/components/cardList/CardList.js:3]: I though react-native-elements dependency was removed! my bad!

File: app/components/loading/Loading.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from "react";
+import { Icon } from "react-native-elements";
+import { View, Text } from "react-native";
+
+import styles from "./styles";
+
+export default class Loading extends Component {
```
Comment (totorototo) [inline app/components/loading/Loading.js:7]: fine!

File: app/routes/screens/main/clubFeed/ClubFeed.js [hunk 4]
```diff
@@ -44,23 +47,9 @@ class ClubFeed extends Component {
...
-    if (isFaulty(club))
-      return (
-        <View style={styles.container}>
-          <Icon name="error" color="#FC4C02" size={100} />
-          <Text style={styles.text}>
-            Oops, I did it again: {getDefect(club)}
-          </Text>
-        </View>
-      );
+    if (isFaulty(club)) return <Faulty message={getDefect(club)} />;
```
Comment (totorototo) [inline app/routes/screens/main/clubFeed/ClubFeed.js:52]: definitely better this way!

**Review by totorototo — APPROVED** [review]
nice job buddy!

File: app/components/cardList/CardList.js (hunk 2 above)
Comment (guipasmoi) [reply app/components/cardList/CardList.js:3]: if its removed it should be removed of package.json

File: app/routes/screens/main/clubFeed/ClubFeed.js (hunk 4 above)
Comment (guipasmoi) [reply app/routes/screens/main/clubFeed/ClubFeed.js:52]: i am considering using recompose https://github.com/acdlite/recompose
to manage component composition
i have to consider slots in this too
i started to do something about theming (light theme/dark theme) and feature flipping
the global thème is how do we compose correctly.

File: app/routes/screens/main/clubFeed/ClubFeed.js (hunk 1 above)
Comment (guipasmoi) [reply app/routes/screens/main/clubFeed/ClubFeed.js:50]: consider this
<Loading progress={club.progress}/>

File: app/routes/screens/main/clubFeed/ClubFeed.js (hunk 4 above)
Comment (guipasmoi) [reply app/routes/screens/main/clubFeed/ClubFeed.js:52]: consider this 
<Faulty message={getDefect(club)} retry={doSomething} />

**Pull request merged**